### ISSUE TITLE
Navigation (Driving/Walking/Bike/Transit) seems to be Google-centric and use outdated API?

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#15978]
Support [Ticket #777801](https://cgeosupport.droescher.eu/scp/tickets.php?id=8395)
cgeo Version: 2024.08.09-RC

A user on support reported the following:
When I select the navigate menu and pick "other external apps" I get to choose WAZE. Cgeo sends the destination coordinates to WAZE and we are off.  However when I select "navigation - drive" I also get the WAZE option. When I click it  WAZE opens and that's the end of that. WAZE doesn't seem to receive the destination (Cgeo doesn't parse the coordinates like in "other apps"). 

I investigated the code for this and discovered something that puzzles me: it seems that the four "standard" navigation options (Navigation (Driving/Walking/Bike/Transit)) are all implemented Google-specific but use an outdated version of their API.
* Currently c:geo sends an intent to an URL like this:  `google.navigation:ll=" + coords.getLatitude() + "," + coords.getLongitude() + "&mode=" + mode)` (where `coords` are the target coords and `mode` specifies whether to navigate by car, bike etc)
* Current documentation for google however claims that URL should loook as follows: `google.navigation:q=latitude,longitude&mode=" + mode`. Also, one shall set a package for the intent: `setPackage("com.google.android.apps.maps")`
* Documentation for Google Map intents: https://developers.google.com/maps/documentation/urls/android-intents

The whole thing looks quite crowded anyway:
![image](https://github.com/user-attachments/assets/d9cbfeb2-0813-4490-9ac3-2932f8041a9c)

This raises several questions:
* is it intended that the "Navigation" intends are Google-specific? If yes, why is there an additional navigation option "Google Maps Directions". I feel this can be simplified....
* If those options are NOT intended to be Google specific, why are they using google-specific Uris in the intend? And why is then there the extra option "Other external app"?

Can someone share light upon why this was implemented the way it currently is? If no one has another opinion I would suggest to make the "Navigation" options Google-specific (by renaming them into "Google Navigation (Driving)" etc), make them comply to current Google API and remove the extra option "Google Maps Directions". Something like this:

![image](https://github.com/user-attachments/assets/56ce04f6-278b-4e5c-a29e-8770b4c6b6f7)

**Comment by eddiemuc** [comment]
Come to think about it: why not reduce this more radically (why should there be different options for Google Maps in the list?):

 
![image](https://github.com/user-attachments/assets/7119fb48-8c90-4265-82c6-8df53babc9db)

**Comment by moving-bits** [comment]
Can't say why it has been implemented this way other than "it's "always" been like this" :-)

But: While I like reducing the number of options, keep in mind that the users can configure which of those options are being displayed to them, so with a more granular list configuration can be adapted better to personal needs.

On the other hand many users will probably never change the defaults, so they will be presented with full options. A different approach therefore could be to reduce activated navigation options in default settings.

Updating the used API seems to be meaningful anyway.

WRT to renaming the menu options: If other apps can hook into the URI used, then we should not rename the menu entries. If the mechanism ensures that only Google Maps will ever be called, then renaming seems to make things clearer.

**Comment by Korb** [comment]
After selecting `Navigation (Walking)` I am prompted to use one of the two mapping apps I have installed and prefer to use over all others - `Magic Earth` and `Here WeGo`. However, after launching both, the waypoint passed to them is not displayed, nor is the route suggested as expected.

The same thing happens when I tap on the compass icon.

---
c:geo 2025.05.14
Magic Earth 
Here WeGo 4.16.100
Magic Earth 7.1.25.18.C9FB8CE8.6141AA41
Android 14

**Comment by moving-bits** [comment]
Does using `Navigation (other)` or `Navigation (other, coords only)` work for you?

**Comment by Korb** [comment]
> Does using `Navigation (other)` or `Navigation (other, coords only)` work for you?

I tried `Navigation (Driving)`, `Navigation (Bike)` and `Navigation (Transit)`, but they behave the same as `Navigation (Walking)`. I specified them in `Default Navigation` [1] and `Secondary Navigation` [2], and activated it by short or long tap on the compass icon. I think this is how it should work.

I haven't tried other options (`Compass`, `Map`, `Other external apps`, `Google Maps Directions` and `Organic Maps`).

I didn't find the `Navigation (other, coords only)` item. I only have `Other external apps (coords only)`. And using it allows to see the coordinates of the place in both applications.

---
_[1] Here you can select your preferred navigation tool.
[2] Here you can select your second preferred navigation tool. Activate it by long tapping the navigation icon next to the title of the cache._

**Comment by moving-bits** [comment]
The `Navigation (Driving)`, `Navigation (Bike)` and `Navigation (Transit)` methods are pretty old methods, implemented specifically to fit the Google Maps API back then.

In contrast to this, the `Other external apps` and `Other external apps (coords only)` (sorry for the naming confusion, my c:geo is set to a different language) use the de facto standard for a coordinates search, adding the title of this waypoint in the first version (which not all apps understand - Here We Go did some time ago, but does no longer).

So if you want to use a navigation app other than Google Maps, it's probably best to deactivate the `Navigation (Driving)`, `Navigation (Bike)` and `Navigation (Transit)` navigation methods by going to settings => routing/navigation => navigation menu, and go with one of the `Other external apps` options.

**Comment by moving-bits** [comment]
I did a quick test after having changed the outdated `ll` parameter to `q`, leaving everything else as is:

Test case is:
- open map
- select a cache
- chose one of the four "navigation: xxx" modes

|navigation app|drive|walk|bike|public transport|
|---|---|---|---|---|
|Google Maps|ok|ok|ok|ok|
|HereWeGo|ok|ok|ok|car mode|
|Waze|destination not detected|destination not detected|destination not detected|destination not detected|
|OsmAnd|car|car|car|car|

So:
- Google Maps works as expected
- HereWeGo does not detect public transport mode and switches to car mode in that case, but everything else works as expected
- OsmAnd detects the destination, but not the mode. But mode switching is a single tap within OsmAnd, so that's ok, I guess.
- Waze detects nothing, neither coordinates nor mode.

I'll prepare a PR to change parameter `ll` to `q`, but leave everything else as is for now.

**Comment by JakeDot** [comment]
Today I stumbled upon another Google "Feature" in the code, that is supposed to open Google Streetview. But it is only active when a "com.google.streetview" app is installed . I can't find any official app for that on the Playstore, is that maybe also obsolete?

**Comment by Korb** [comment]
> `com.google.streetview`

https://web.archive.org/web/20250000000000*/https://play.google.com/store/apps/details?id=com.google.streetview — N/A

https://reports.exodus-privacy.eu.org/en/reports/com.google.streetview/latest/ — N/A

🤔

**Comment by JakeDot** [comment]
sorry, it's actually called ["com.google.android.street"](https://play.google.com/store/apps/details?id=com.google.android.streetview) however also nothing there, I think it has been replaced by Google Earth

**Comment by moving-bits** [comment]
yeah, we got the deprecation notice back then but decided to leave that in as well, for those users that still had installed that app.
